### ISSUE TITLE
More fluffy craftiness!

### ISSUE THREAD

**Issue by craftoid** [forresto/turtle-svg#10]
You mentioned that you were planning to do some embroidery stuff...
Are you planning to add this to the laser turtle?

Embroidery is really what I would be most interested in.
I already created a fork over at bitcraft for my wooly explorations:
https://github.com/bitcraftlab/turtlecraft

Should I try to keep it in sync with turtle-svg, so you can merge stuff back in?

**Comment by forresto** [comment]
I'm happy to merge stuff back and forth. 

I don't have access to an embroidery machine, but a friend has an industrial one a couple hours away. Originally I was thinking that I'd import the SVG to the software that he uses to get it ready. But it would be nice to find one closer to experiment with more directly, with each `f()` a stitch. How is PES?

Part of this was trying out techniques to make a live coding canvas module for Meemoo. Still have to think about how to eval that code.

**Comment by craftoid** [comment]
PES is pretty popular with the home embroidery scene. It's the format used by brother embroidery machines.
And there's lots of open source and freeware stuff for showing thumbnails in MS Explorer / Mac Finder etc...
They got a Brother Embroidery machine in a makerspace nearby, so I can try it out...

**Comment by forresto** [comment]
Can you define each stitch?

I guess it wouldn't be too hard to parse the generated `M 0 0 l 100 100` into any number of different formats... PES, PNG, G-code... At my hackerspace I have to import the SVG to Corel Draw before printing to the Epilog. Addie from http://labs.nortd.com/lasersaur/ is working on direct turtle-svg to laser cutter, which seems more open-pure :wink: 

One experiment I have in mind is printing 3D forms with different kinds of textures, with the turtle directly controlling the print head. Will probably end up with some broken plastic, but potentially come interesting textures. And how to visualize that? 

There is also http://web.media.mit.edu/~ericr/beetleblocks.php pointed out on Twitter... I'd like to play with a Blockly + Three.js version of this. How is the Blockly hacking going?

**Comment by craftoid** [comment]
You can also use different colors/threads, using a standardized color palette.
I think I'll just create a custom worker for PES output. This would probably require some logic to switch between workers, but seems to make a lot of sense.

I already toyed around with a Hand-Embroidery SVG worker (still need to integrate it).

BeetleBlocks looks like fun! Blockly + ThreeJS would sure be awesome.
I was also thinking about creating a Blockly-powered Shadertoy  ...

Work on **Processing Blocks** is just getting started.
It will be a Java extension for the Processing IDE using OpenBlocks.
After that I'll tackle **Processing Blocks JS**. 
This will be a Blockly based Webapp to create + share those Block Programs online ...
The idea is to have a common file format for Block Languages + Block Programs,
that can both be used by Processing Blocks and Processing Blocks JS.

So many ideas, so little time ;-)

**Comment by forresto** [comment]
Re: ideas/time, don't I know it.

If you can embroider [this](http://forresto.github.com/turtle-svg/#code/jY/NbsIwEITvfooRJyMBCVBaVAQnjhx5AZNugiVjV2s7HBDvjklApD+ove3MfpqdzTKsWR2hwFRE9rom+E9V0LDUxmhbIbk1vYtaMTxVB7JhQ7YKeywxni1EGW0RtLOIVgepA7G6yj5OAtAlHhZWyFsb32AMMe4vmgXLfDSZ3cRTqpRfqvyF/yf0LM7i8cxx7wwlyreFQ2S7dXKazRv2FuEb8VuXDpBSC2ccyx7TRy8ZB1fTNWs6wCRPuj31mqY7WDGR7aAvb0/RnYnUJfMf5AU=) in three colors I'd be really happy to see it.
